src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -214,21 +219,22 @@ public void setArray(int parameterIndex, Array x) throws SQLException {
     public void setDate(int parameterIndex, Date x, Calendar cal) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
-        throw new SQLFeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+        throw new SQLFeatureNotSupportedException("Date type is not supported");
     }
 
     @Override
     public void setTime(int parameterIndex, Time x, Calendar cal) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
-        throw new SQLFeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+        throw new SQLFeatureNotSupportedException("Time type is not supported");
     }
 
     @Override
-    public void setTimestamp(int parameterIndex, Timestamp x, Calendar cal) throws SQLException {
+    public void setTimestamp(int parameterIndex, Timestamp timestamp, Calendar calendar) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
-        throw new SQLFeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+        checkTimeZone(calendar.getTimeZone());
+        setParameter(parameterIndex, toBsonValue(timestamp));
```

Comment:
There is a way to make Hibernate ORM to use `setObject`/`getObject` for `Instant`, thus saving us from dealing with `Timestamp` and `Calendar`, without dealing with the `JAVA_TIME_USE_DIRECT_JDBC`, `PREFERRED_INSTANT_JDBC_TYPE` configuration properties. I propose us to use it, as it makes the implementation and even testing simpler.

The proposed change is done in https://github.com/stIncMale/mongo-hibernate/commit/5bebe91ace03925b065e3f13c0985329c315c699.